test/Test/Toml/Gen.hs

```diff
@@ -150,9 +151,55 @@ genZoned = do
     pure $ ZonedTime local zTime
 
 genDate :: MonadGen m => m DateTime
-genDate = Gen.choice 
+genDate = Gen.choice
     [ Day   <$> genDay
     , Hours <$> genHours
     , Local <$> genLocal
     , Zoned <$> genZoned
-    ]
\ No newline at end of file
+    ]
+
+genBool :: MonadGen m => m Bool
+genBool = Gen.bool
+
+genInteger :: MonadGen m => m Integer
+genInteger = toInteger <$> Gen.int (Range.constantBounded @Int)
+
+genDouble :: MonadGen m => m Double
+genDouble = Gen.double $ Range.constant @Double (-1000000.0) 1000000.0
+
+genText :: MonadGen m => m Text
+genText = Gen.text (Range.constant 0 256) Gen.alphaNum
+
+genListBoolR :: MonadGen m => m [Value 'TBool]
+genListBoolR = Gen.recursive Gen.choice [genListBool] [genListBoolR]
+  where 
+    genListBool = Gen.list (Range.constant 1 10) $ Bool <$> genBool
+
+genListIntegerR :: MonadGen m => m [Value 'TInteger]
+genListIntegerR = Gen.recursive Gen.choice [genListInteger] [genListIntegerR]
+  where 
+    genListInteger = Gen.list (Range.constant 1 10) $ Integer <$> genInteger
+
+genListDoubleR :: MonadGen m => m [Value 'TDouble]
+genListDoubleR = Gen.recursive Gen.choice [genListDouble] [genListDoubleR]
+  where 
+    genListDouble = Gen.list (Range.constant 1 10) $ Double <$> genDouble
+
+genListTextR :: MonadGen m =>  m [Value 'TText]
+genListTextR = Gen.recursive Gen.choice [genListText] [genListTextR]
+  where 
+    genListText = Gen.list (Range.constant 1 10) $ Text <$> genText
+
+genListDateR :: MonadGen m => m [Value 'TDate]
+genListDateR = Gen.recursive Gen.choice [genListDate] [genListDateR]
+  where
+    genListDate = Gen.list (Range.constant 1 10) $ Date <$> genDate
```

Comment:
Current implementation is too verbose and contains a lot of boilerplate. I propose the following solution:

1. Have `genNonArray` function which is like `genAnyValue` but without `genArray`.
2. Implement `genArray` function (using `genNonArray` generator) that generates list of `AnyValue` and then unsafely typechecks this list and creates `Array`. You should use `recursive` combinator for this.
3. Implemeng `genAnyValue` which is a combination of `genNonArray` and `genArray`.

Reply:
So, basically you now have more code than needed. And you don't use `recursive` in `genArray` (thus you can't generate nested arrays yet).

I imagine the following implementation:

```haskell
genArray = Gen.recursive Gen.choice [genArrayFromAnyValueList] [genArray]
```